File: src/main/java/ir/reservs/reservs/ui/main/settings/SettingsContract.java
package ir.reservs.reservs.ui.main.settings;

import ir.reservs.reservs.ui.base.BaseFragmentContract;

public class SettingsContract {
    interface View extends BaseFragmentContract.View {
        void setUserInfo(String name, String phone, String image);

        void openLoginActivity();
    }

    interface Presenter extends BaseFragmentContract.Presenter<View> {
        void logoutUser();
    }
}

File: src/main/java/ir/reservs/reservs/ui/main/settings/SettingsContract.kt
package ir.reservs.reservs.ui.main.settings;

import ir.reservs.reservs.ui.base.BaseFragmentContract;

class SettingsContract {
    interface View : BaseFragmentContract.View {
        fun setUserInfo(name: String, phone: String, credit: Long)
        fun setUserImage(image: String)
        fun openLoginPage()
        fun showProgress()
        fun hideProgress()
    }

    interface Presenter : BaseFragmentContract.Presenter<View> {
        fun logoutUser()
    }
}
